jscomp/others/js_array.ml

```diff
@@ -17,61 +17,122 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
-
 type 'a t = 'a array
+type 'a array_like
```

Comment:
These were added primarily for the iterator functions, and if they're taken out there's currently no use for them. But there are many objects elsewhere that are array-like, `arguments`, `NodeList` and `HTMLCollection` to name a few. Without `array_like` and the `Array.from` function, these would all have to implement `Array.from` themselves as `toArray`. Unsure which is the right approach here.